plugins/modules/ec2_spot_instance.py

```diff
@@ -0,0 +1,359 @@
+      placement_group:
+        description:
+          - Placement group for the instance.
+        type: str
+      ramdisk:
+        description:
+          - Ramdisk eri to use for the instance.
+        type: str
+      monitoring:
+        description:
+          - Enable detailed monitoring (CloudWatch) for the instance.
+        type: bool
+        default: false
+  price:
+    description:
+      - Maximum spot price to bid. If not set, a regular on-demand instance is requested.
+      - A spot request is made with this maximum bid. When it is filled, the instance is started.
+    type: str
+  request_type:
+    description:
+      - The type of spot request.
+      - After being interrupted a C(persistent) spot instance will be started once there is capacity to fill the request again.
+    default: "one-time"
+    choices: [ "one-time", "persistent" ]
+    type: str
+  state:
+    description:
+      - When I(state=present), I(launch_specification) is required.
+      - When I(state=absent), I(spot_instance_request_ids) is required.
+    default: 'present'
+    choices: ['absent', 'present']
+    type: str
+  spot_price:
+    description:
+      - Maximum spot price to bid. If not set, a regular on-demand instance is requested.
+      - A spot request is made with this maximum bid. When it is filled, the instance is started.
+    type: str
+  spot_type:
+    description:
+      - The type of spot request.
+      - After being interrupted a C(persistent) spot instance will be started once there is capacity to fill the request again.
+    default: "one-time"
+    choices: [ "one-time", "persistent" ]
+    type: str
+  valid_from:
+    description:
+      -  The start date of the request. If this is a one-time request, the request becomes active at this date and time and remains active until all instances launch, the request expires, or the request is canceled.
+      -  If the request is persistent, the request becomes active at this date and time and remains active until it expires or is canceled.      
+  valid_until:
+    description:
+      - The end date of the request
+  tags:
+    description:
+      - tag:value pairs to add to the volume after creation.
+    default: []
+    type: list
+  spot_instance_request_ids:
+    description:
+        - List of strings with IDs of spot requests to be cancelled
+    default: []
+    type: list
+'''
+
+EXAMPLES = '''
+# Simple Spot Request Creation
+- hosts: localhost
+  tasks:
+    - name: Test EC2 spot instance module
+      amazon.aws.ec2_spot_instance:
+        launch_specification:
+          image_id: "ami-123456789"
+          key_name: "my-keypair"
+          instance_type: "t2.medium"
+
+# Simple Spot Request Termination
+- hosts: localhost
+  tasks:
+    - name: Test EC2 spot instance cancel module
+      amazon.aws.ec2_spot_instance:
+        spot_instance_request_ids: ['sir-d8468pbj', 'sir-qph6aytk']
+        state: 'absent'
+'''
+
+RETURN = '''
+spot_request_info:
```

Comment:
```suggestion
spot_request:
```
_info is generally going to be associated by Ansible users as relating to an *_info.py module. Plus it's nice to keep these keys shorter, to save folks some typing.